src/NuGet.Clients/NuGet.PackageManagement.UI/Options/CheckedListBoxAccessibleObject.cs

```diff
@@ -31,7 +32,16 @@ public override AccessibleObject GetChild(int index)
         {
             if (index >= 0 && index < CheckedListBox.Items.Count)
             {
-                var packageSource = (PackageSourceContextInfo)CheckedListBox.Items[index];
+                var item = (PackageSourceContextInfo)CheckedListBox.Items[index];
+                PackageSource packageSource = new PackageSource(item.Source, item.Name);
+                if (packageSource.IsHttp && !packageSource.IsHttps)
+                {
+                    var sourceMessage = string.Concat(
+                        "Warning: Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.",
```

Comment:
This needs to be in a string resource, so it can be localized for the customer's language.